### DISCUSSION TITLE
Some question about using it on CF and KV

### DISCUSSION THREAD

**Discussion by A5yncX** [dddddddddzzzz/OpenHeart#8] [Questions] [answered]
I linked KV, and I tried to copy worker.js to the CF, but it always return the error page.

<img width="563" alt="image" src="https://github.com/user-attachments/assets/01484878-30d3-45f0-9e00-8a6fca73d7a0">
<img width="570" alt="image" src="https://github.com/user-attachments/assets/641a52e2-5dc0-430c-824a-b42d16a3de16">

**Comment by muan** [comment, marked as answer]
On your L17 and L18, the `NAMESPACE` variable is underlined indicating they're missing. You might need to add `bindings` to your worker.

Go to Workers & Pages → your Worker → Settings → Bindings → + Add → KV namespace → `Variable name` would be the `NAMESPACE` variable name and KV namespace being the KV you've set up (via Workers & Pages → KV) . 

Say you can name it `OPENHEART` then L18 becomes `OPENHEART.put(key, ...)`.

This should fix it.

---

For example, this is my Bindings setup, the variable name is `KV`:

<img width="775" alt="image" src="https://github.com/user-attachments/assets/a3f81956-2745-4a50-8ce2-b84c4af7ea4a">

And this is the code (your L17-L18):

```js
  const currentCount = Number(await(KV.get(key)) || 0)
  await KV.put(key, currentCount + 1)
```

**Reply by A5yncX** [reply]
Thanks, now it's this:
<img width="783" alt="image" src="https://github.com/user-attachments/assets/c177c14b-cd30-48d4-ba74-026b8006a09f">
<img width="728" alt="image" src="https://github.com/user-attachments/assets/dbecea32-5381-49dd-9ac2-7265007cdabb">
but still the same error😂

**Reply by muan** [reply]
Ah you're right, looks like a TypeScript error instead.
 
But does your request still fail?

For example, do something like this:

![](https://github.com/user-attachments/assets/df9e518a-36f0-42da-916c-a00a340d3111)

**Reply by A5yncX** [reply]
That's worked! and which link is my API link? Do I need to include `?id=test`?
<img width="540" alt="image" src="https://github.com/user-attachments/assets/8b6049ee-242a-4175-9676-f82ad397f169">

**Reply by muan** [reply]
`https://likepost.asyncx0806.workers.dev/` is the URL assigned to your worker by Cloudflare.
You can add custom routes and disable this default if you want via Worker → Domains & Routes.

In this example code I provided, `id` param is the ID of the thing that you are emoji reacting to. 

So with this now set up, you should be able to make this request one the command line:

```
curl -d '<emoji>' -X POST 'https://likepost.asyncx0806.workers.dev/?id=<mywebsite>'
```

And this demo code does not implement the GET request (where you can get the reactions for IDs).
You can see the reactions without the API in Worker → KV → likepost → View → KV Pairs, like this:

<img width="807" alt="" src="https://github.com/user-attachments/assets/13be5c8f-f9c9-4525-ae1e-c6cc9fdefe49">

Here the IDs I use is the path of the URLs on my website, and the emoji is always ❤️.

**Comment by muan** [comment]
So the example code is only a starting point

To help you a bit further, these are the code do implement the GET request to look up the count:

L1, going from only handling POST request, to separating POST and GET:

```diff
addEventListener("fetch", event => {
+ if (event.request.method === 'GET') {
+   event.respondWith(handleGET(event.request))
+ }
+ if (event.request.method === 'POST') {
+   event.respondWith(handlePOST(event.request))
+ }
- event.respondWith(handleRequest(event.request))
})
```

And adding this function to handle GET request:

```js
async function handleGET(request) {
    const url = new URL(request.url)
    const id = url.searchParams.get('id')
    if (!id) {
      return new Response(0, {headers: getHeaders})
    }

    const list = (await SITE.list({prefix: `${id}::`}))
    const keys = list.keys.map(e => e.name)
    const json = {}
    for (const key of keys) {
      const emoji = key.split('::')[1]
      json[emoji] = await SITE.get(key) || 0
    }

    return new Response(JSON.stringify(json), {headers: getHeaders})
}
```

Renaming the POST function:

```diff
+ async function handlePOST(request) {
- async function handleRequest(request) {
```

**Comment by A5yncX** [comment]
I have customized the domain name for it, it's `likepost.asyncx.top`

I tried to use this command, and it return `ok`.
```
curl -d '😂' -X POST 'https://likepost.asyncx.top/?id=blog.asyncx.top'
```
I changed the `worker.js` code by you provided in the previous message.

If possible, you can access my blog by `blog.asyncx.top`

I've add this code for my website:
<img width="765" alt="image" src="https://github.com/user-attachments/assets/edae62d6-c82b-447e-bc28-33773a87a54b">

**Reply by muan** [reply]
Two things,

1. you need an ID in the form URL: `?id=<insert id here>`. It currently responses `not ok` because of the missing ID param.
2. If you want to change the behavior of returning `ok`, you'd need to change the code at the end of `handlePost`, 

For example, in our public API we allow you to choose to redirect back with [these lines of code](https://github.com/dddddddddzzzz/api-oh/blob/312d490641c7ec76ff02e9981481ae1469131075/src/worker.js#L104-L105).
